### DISCUSSION TITLE
Internet Exchanges - Add/Import Issues

### DISCUSSION THREAD

**Discussion by moitrierk** [peering-manager/peering-manager#632] [Q&A] [answered]
Hello,

I'm trying to add new peers from the Internet Exchanges webpage, but I'm currently facing on several issues.

#1: Import Internet Exchanges from PeeringDB :

Peering DB is sync, but when I tried to Import Internet Exchanges from PeeringdB, I got the error below : 

There was a problem with your request. Please contact an administrator.
The complete exception is provided below:
<class 'ValueError'>
not enough values to unpack (expected 2, got 1)

Python version: 3.9.5
Peering Manager version: v1.7.2

-----------

#2: Add available peers from Internet Exchanges

As the import from peeringdB doesn't work, I created manually the IX AMS-IX.
I created an interconnection with my router, and I see the available peers.
But when I tried to add one of them, it doesn't work.
I tried to collected some logs to try and see where is the error, but without success.
Just see below the logs when I clicked on the Buttom Add Selected :

xxxx - - [29/Sep/2022:14:18:55 +0000] "POST /internet-exchange-peering-sessions/add_from_peeringdb/ HTTP/1.1" 200 12167 "http://xxxxx:8080/internet-exchanges/6/peers/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:105.0) Gecko/20100101 Firefox/105.0"
xxxx - - [29/Sep/2022:14:18:56 +0000] "POST /internet-exchange-peering-sessions/add_from_peeringdb/ HTTP/1.1" 302 0 "http://xxxxx:8080/internet-exchange-peering-sessions/add_from_peeringdb/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:105.0) Gecko/20100101 Firefox/105.0"
xxxx - - [29/Sep/2022:14:18:58 +0000] "GET /internet-exchanges/6/peers/ HTTP/1.1" 200 402803 "http://xxxxx:8080/internet-exchange-peering-sessions/add_from_peeringdb/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:105.0) Gecko/20100101 Firefox/105.0"

#3: No preferred affiliated autonomous system selected

As you can see in one the images included, I have this warning message : No preferred affiliated autonomous system selected
Whereras I selected my autonomous system 56665 (Affiliated) in the Autonomous System page.

![image](https://user-images.githubusercontent.com/6058000/193237657-0835104c-adc6-4504-9662-9d61ae36228c.png)
![image](https://user-images.githubusercontent.com/6058000/193237118-67a1bbca-52c0-42dd-b40b-547811bdd5c2.png)
![image](https://user-images.githubusercontent.com/6058000/193237200-67f0ed61-873f-49ed-b32c-4b2957f5f7e1.png)

Many thanks in advance for your help and support.

Best regards,
Kevin

**Comment by gmazoyer** [comment, marked as answer]
If you still have the "No preferred affiliated autonomous system selected" header, it means you haven't selected any yet. You'll need to select one from the list on [the right of the top bar](https://peering-manager.readthedocs.io/en/stable/user-interface/#user-profile).

Note that only autonomous systems with the "affiliated" field set will appear in the list.
